LenovoLegionToolkit.Cmdline/Program.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using LenovoLegionToolkit.Cmdline.Resources;
+using LenovoLegionToolkit.Cmdline.Utils;
+using LenovoLegionToolkit.Lib.Automation;
+using LenovoLegionToolkit.Lib.Automation.CmdLine;
+using LenovoLegionToolkit.Lib.Extensions;
+
+namespace LenovoLegionToolkit.Cmdline;
+
+public class Program
+{
+    public static async Task<int> Main(string[] args)
+    {
+        var flags = new Flags(args);
+        if (flags.Error)
+        {
+            Console.WriteLine(Resource.Error_IllegalCommandLineArgument_Text);
+            return 1;
+        }
+
+        await LocalizationHelper.SetLanguageAsync();
+
+        if (flags.ShowHelp)
+        {
+            ShowHelpMessage();
+            return 0;
+        }
+
+        if (flags.QuickActionRunName is not null)
+        {
+            await ExecuteQuickActionRunAsync(flags.QuickActionRunName ?? string.Empty, flags.Silent);
+            return 0;
+        }
+
+        if (!flags.Silent)
+        {
+            Console.WriteLine(Resource.Error_IllegalCommandLineArgument_Text);
+        }
+        return 1;
+    }
+
+    private static void ShowHelpMessage()
+    {
+        Console.WriteLine($"llt.exe - {Resource.HelpMessage_ExeDescription}");
+        Console.WriteLine($"{Resource.HelpMessage_Version} {GetVersion()}\n");
+        Console.WriteLine($"{Resource.HelpMessage_AvailableArguments}");
+        Console.WriteLine($" * --help\t\t{Resource.HelpMessage_Argument_Help}");
+        Console.WriteLine($" * --silent\t\t{Resource.HelpMessage_Argument_Silent}");
+        Console.WriteLine($" * --run={{Quick Action}}\t{Resource.HelpMessage_Argument_Run}\n");
+    }
+
+    private static string GetVersion()
+    {
+        var version = Assembly.GetEntryAssembly()?.GetName().Version;
+        if (version is null)
+        {
+            return string.Empty;
+        }
+        else if (version.IsBeta())
+        {
+            return "BETA";
+        }
+        else
+        {
+            return version.ToString(3);
+        }
+    }
+
+    private static async Task ExecuteQuickActionRunAsync(string quickAcionName, bool silent)
+    {
+        var client = new CmdLineIPCClient();
+        await client.RunQuickActionAsync(quickAcionName);
+
+        if (!silent)
+        {
+            if (client.State == CmdLineQuickActionRunState.ActionRunFailed)
```

Comment:
This many `if` statements would look better if replace with a `switch` statement.